scopes/dependencies/pnpm/lockfile-deps-graph-converter.ts

```diff
@@ -244,6 +268,38 @@ export function convertGraphToLockfile(
       }
     }
   }
+  const pkgsToResolve: Array<{ name: string; version: string; pkgId: string }> = [];
+  const pkgsInTheWorkspaces = new Set(Object.values(manifests).map(({ name }) => name));
+  for (const [pkgId, pkg] of Object.entries(lockfile.packages)) {
+    if (pkg.resolution == null || 'type' in pkg.resolution && pkg.resolution.type === 'directory') {
+      const parsed = dp.parse(pkgId)
+      if (parsed.name && parsed.version && !pkgsInTheWorkspaces.has(parsed.name)) {
+        pkgsToResolve.push({
+          name: parsed.name,
+          version: parsed.version,
+          pkgId,
+        })
+      }
+    }
+  }
+  await Promise.all(pkgsToResolve.map(async (pkgToResolve) => {
+    if (lockfile.packages[pkgToResolve.pkgId].resolution == null) {
+      const { resolution } = await resolve({
+        alias: pkgToResolve.name,
+        pref: pkgToResolve.version,
+      }, {
+        lockfileDir: '',
+        projectDir: '',
+        registry: pickRegistryForPackage(registries, pkgToResolve.name),
+        preferredVersions: {},
+      })
+      if ('integrity' in resolution && resolution.integrity) {
```

Comment:
I think you don't need the first part. it's enough to do `if (resolution.integrity)`

Reply:
It is not enough because it is a union of types and not each has "integrity". Alternatively I could probably check the value of the "type" field. That helps typescript to assert the right type from the union.